overcloud.yml

```diff
@@ -133,12 +133,30 @@
             infrared tripleo-overcloud -vvv --version {{ osp_release }} --deployment-timeout 240 --deployment-files {{ nic_configs }} --introspect no --tagging no --deploy yes --controller-nodes {{ controller_count }} --compute-nodes {{ compute_count }} --network-protocol ipv4 --network-backend {{ network_backend }} {{ network_type }} true --public-network false > {{ log_directory }}/overcloud_deploy.log 2>&1
         args:
             chdir: "{{ infrared_dir }}"
-        when: extra_templates is not defined and parameter_defaults is not defined
+        when: tls_everywhere != true and extra_templates is not defined and parameter_defaults is not defined
+
+      - name: run tripleo-overcloud deploy tls
+        shell: |
+            source .venv/bin/activate
+            infrared tripleo-overcloud -vvv --version {{ osp_release }} --deployment-files {{ nic_configs }} --introspect no --tagging no --deploy yes --controller-nodes {{ controller_count }} --compute-nodes {{ compute_count }} --tls-everywhere true > {{ log_directory }}/overcloud_deploy.log 2>&1
+        args:
+            chdir: "{{ infrared_dir }}"
+        when: tls_everywhere == true and extra_templates is not defined and parameter_defaults is not defined
+
 
       - name: run tripleo-overcloud deploy with extra templates
         shell: |
             source .venv/bin/activate
             infrared tripleo-overcloud -vvv --version {{ osp_release }} --deployment-timeout 240 --build {{ osp_puddle }}  --deployment-files {{ nic_configs }} --introspect no --tagging no --deploy yes --controller-nodes {{ controller_count }} --compute-nodes {{ compute_count }} --overcloud-templates {{ infrared_dir }}/plugins/tripleo-overcloud/vars/overcloud/templates/extra.yml --network-protocol ipv4 --network-backend {{ network_backend }} {{ network_type }} true --public-network false > {{ log_directory }}/overcloud_deploy.log 2>&1
         args:
             chdir: "{{ infrared_dir }}"
-        when: ( extra_templates is defined and extra_templates|length>0 ) or ( parameter_defaults is defined and parameter_defaults|length>0 )
+        when: tls_everywhere !=true and ( extra_templates is defined and extra_templates|length>0 ) or ( parameter_defaults is defined and parameter_defaults|length>0 )
```

Comment:
Can also use the "not" notation like not tls_everywhere

Reply:
So when we set tls_everywhere to true in group_vars/all.yml. Then not tls_everywhere will result it into false